core/src/main/java/org/apache/brooklyn/location/byon/ByonLocationResolver.java

```diff
@@ -152,49 +153,54 @@ protected ConfigBag extractConfig(Map<?,?> locationFlags, String spec, LocationR
         
         String osFamily = (String) machineConfig.remove(OS_FAMILY.getName());
         String ssh = (String) machineConfig.remove("ssh");
-        String winrm = (String) machineConfig.remove("winrm");
-        Map<Integer, String> tcpPortMappings = (Map<Integer, String>) machineConfig.get("tcpPortMappings");
-        
-        checkArgument(ssh != null ^ winrm != null, "Must specify exactly one of 'ssh' or 'winrm' for machine: %s", valSanitized);
-        
-        UserAndHostAndPort userAndHostAndPort;
-        String host;
-        int port;
-        if (ssh != null) {
-            userAndHostAndPort = parseUserAndHostAndPort(ssh, 22);
+        if (machineConfig.containsKey("winrm") && !(machineConfig.get("winrm") instanceof String)) {
```

Comment:
Looks good.
@iyovcheva what do you think about supporting external provider for ssh address as well. It will be more consistent.